modules/reactor/src/meshgenerators/PlaneIDGenerator.C

```diff
@@ -0,0 +1,115 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PlaneIDGenerator.h"
+#include "CastUniquePointer.h"
+
+#include "libmesh/elem.h"
+
+registerMooseObject("ReactorApp", PlaneIDGenerator);
+
+defineLegacyParams(PlaneIDGenerator);
+
+InputParameters
+PlaneIDGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<std::vector<Real>>("z_layers", "Z coordinates of axial layers");
+  params.addParam<std::vector<unsigned int>>("z_sublayers",
+                                             "The number of layers for each z-layers");
+  params.addRequiredParam<std::string>("id_name", "Name of Integer ID set");
+  params.addClassDescription(
+      "This PlaneIDGenerator source code is to assigns plane reporting ID for existing 3D meshes");
+
+  return params;
```

Comment:
this is very specific (Z planes, only planes supported etc), lets have something more general

Reply:
`PlaneIDGenerator` was updated to take `x`, `y` or `z` as a plane axis option as:

- Example 1
```
  [pin3d_id]
    type = PlaneIDGenerator
    input = 'pin3d'
    layer_coordinates = '0.0 5.0 10.0 15.0'
    num_sublayers = ' 1 2 1'
    layer_axis = 'z'
    id_name = 'plane_id'
  []
```

- Example 2
```
  [grid]
    type = CartesianMeshGenerator
    dim = 2
    dx = '5.0'
    ix = '2'
    dy = '5.0 10.0 '
    iy = '1 4'
  []

  [plane_id_gen]
    type = PlaneIDGenerator
    input = 'grid'
    layer_coordinates = '0.0 5.0 15.0'
    num_sublayers = ' 1 2'
    layer_axis = 'y'
    id_name = 'plane_id'
  []
```